laravel/resources/js/ui/schedule.js

```diff
@@ -67,6 +67,19 @@ $(document).ready(function()
         }
     });
 
+    //format duration fields on the fly when the server can't
+    $('#duration-field').on('change', function()
+    {
+        if ($(this).value() == 'custom')
+        {
+            var currentVal = $('input[name=custom_duration]').value();
+            if(currentVal.length<5)
```

Comment:
All of this logic (line 76-79) can be replaced by https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/padStart

Reply:
And just to clarify, I wasn't advocating managing this in PHP, I linked my previous commit to show the backend behavior that remediated the issue. JS is almost certainly the right place to fix this, I'm just super rusty so I didn't undertake it.